go-controller/pkg/node/management-port-dpu.go

```diff
@@ -47,15 +40,10 @@ func (mp *managementPortRepresentor) Create(nodeAnnotator kube.Annotator, waiter
 	// Get management port representor netdevice
 	link, err := util.GetNetLinkOps().LinkByName(mp.repName)
 	if err != nil {
-		if !util.GetNetLinkOps().IsLinkNotFoundError(err) {
-			return nil, fmt.Errorf("failed to lookup %s link: %v", mp.repName, err)
-		}
-		// It may fail in case this is not the first run after reboot and management port has already been renamed.
-		link, err = util.GetNetLinkOps().LinkByName(k8sMgmtIntfName)
-		if err != nil {
-			return nil, fmt.Errorf("failed to get link device for %s. %v", mp.repName, err)
-		}
-	} else if mp.repName != k8sMgmtIntfName {
+		return nil, err
+	}
+
+	if link.Attrs().Name != k8sMgmtIntfName {
```

Comment:
Do we actually need this anymore? Since mp.repName should have already went through `getManagementPortNetDev()`. I feel at this point link.Attrs().Name  should equal mp.repName.

Reply:
Thinking about it more, I think it should be fine as is. We will use the `link.Attrs().Name` just in case.

Reply:
1. `link.Attrs().Name` may not be equal to `mp.repName` if altnames are used.
2. We need this check to do a lookup for existing mgmt port only when new mgmt doesn't have name of `ovn-k8s-mp*`.